packages/flutter_tools/lib/src/web/devfs_web.dart

```diff
@@ -256,18 +247,110 @@ class WebAssetServer {
     }
     return modules;
   }
+
+  // Attempt to resolve `path` to a dart file.
+  File _resolveDartFile(String path) {
+    // If this is a dart file, it must be on the local file system and is
+    // likely coming from a source map request. Attempt to look in the
+    // local filesystem for it, and return a 404 if it is not found. The tool
+    // doesn't currently consider the case of Dart files as assets.
+    File file = globals.fs.file(globals.fs.currentDirectory.uri.resolve(path));
+    final List<String> segments = path.split('/');
+    if (segments.first.isEmpty) {
+      segments.removeAt(0);
+    }
+    // If both of the lookups above failed, the file might have been a package
```

Comment:
nit: which two lookups is this referring to?

Reply:
Done